src/EventTimer.ts

```diff
@@ -32,24 +43,26 @@ export class EventTimer {
 	 * Note: We save to window.guardian.commercialTimer because
 	 * different bundles (DCR / DCP) can use commercial core, and we want
 	 * all timer events saved to a single instance per-page
-	 *
+	 * @param {string} trackerName - Name of the GA tracker
 	 * @returns {EventTimer} Instance of EventTimer
 	 */
-	static init(): EventTimer {
-		return (window.guardian.commercialTimer ||= new EventTimer());
+	static init(gaTrackerName: string): EventTimer {
+		return (window.guardian.commercialTimer ||= new EventTimer(
+			gaTrackerName,
+		));
```

Comment:
What happens if you try to init a second time with a different `gaTrackerName`? It just uses the previous singleton, right?

Reply:
Yeah I kind of broke the singleton in that way since it accepts params. In reality we wouldn't call it twice with different param but it does break the design pattern.
I've added it in as param because we haven't migrated [config lib](https://github.com/guardian/frontend/blob/9a89d67f4c7d867f0dc5dcafdb940fbc7e1dfb3c/static/src/javascripts/lib/config.js#L8) from frontend into commercial core yet, but I guess I _could_ just directly get it from the window without going via the lib.

Reply:
I've changed it to get the tracker name from the window object directly instead of passing it as a param inside the init